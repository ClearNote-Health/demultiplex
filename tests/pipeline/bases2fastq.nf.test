nextflow_pipeline {

    name "Test Workflow main.nf - BASES2FASTQ"
    script "main.nf"
    profile "test_bases2fastq"
    tag "bases2fastq"
    tag "pipeline"

    test("Bases2Fastq") {

        when {
            params {
                outdir        = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert workflow.trace.succeeded().size() == 8 },
                { assert snapshot(
                    // FIXME
                    // path("$outputDir/sim-data/DefaultSample_R1.fastq.gz.md5"),
                    // path("$outputDir/sim-data/DefaultSample_R2.fastq.gz.md5"),
                    path("$outputDir/sim-data/Metrics.csv"),
                    path("$outputDir/sim-data/UnassignedSequences.csv"),
                    path("$outputDir/sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_R1.fastq.gz"),
                    path("$outputDir/sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_R2.fastq.gz")
                ).match("bases2fastq") },
                { assert new File("$outputDir/sim-data/20230404-Bases2Fastq-Sim_QC.html").exists() },
                { assert new File("$outputDir/sim-data/RunStats.json").exists() },
                { assert new File("$outputDir/sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_stats.json").exists() },
                { assert new File("$outputDir/sim-data/RunManifest.json").exists() }
            )
        }

    }

}
