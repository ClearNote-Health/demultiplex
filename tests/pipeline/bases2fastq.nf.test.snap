{
    "Bases2Fastq": {
        "content": [
            8,
            {
                "BASES2FASTQ": {
                    "bases2fastq": "1.8.0.1260801529, use subject to license available at elementbiosciences.com"
                },
                "FALCO": {
                    "falco": "1.2.1"
                },
                "FASTP": {
                    "fastp": "0.23.4"
                },
                "MD5SUM": {
                    "md5sum": 8.3
                },
                "UNTAR_FLOWCELL": {
                    "untar": 1.34
                },
                "Workflow": {
                    "nf-core/demultiplex": "v1.5.2"
                }
            },
            [
                "fastq",
                "fastq/DefaultSample.samplesheet.csv",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/fastp-insert-size-plot.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-gc-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-content-n-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_1_Before_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_2_After_filtering.txt",
                "multiqc/multiqc_data/fastp-seq-quality-plot_Read_2_Before_filtering.txt",
                "multiqc/multiqc_data/fastp_filtered_reads_plot.txt",
                "multiqc/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/multiqc_data/fastqc_sequence_length_distribution_plot.txt",
                "multiqc/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_fastp.txt",
                "multiqc/multiqc_data/multiqc_fastqc.txt",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/fastp-insert-size-plot.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-gc-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-content-n-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_1_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_1_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_2_After_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp-seq-quality-plot_Read_2_Before_filtering.pdf",
                "multiqc/multiqc_plots/pdf/fastp_filtered_reads_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastp_filtered_reads_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_sequence_length_distribution_plot.pdf",
                "multiqc/multiqc_plots/pdf/fastqc_top_overrepresented_sequences_table.pdf",
                "multiqc/multiqc_plots/pdf/general_stats_table.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/fastp-insert-size-plot.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-gc-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-content-n-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_1_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_1_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_2_After_filtering.png",
                "multiqc/multiqc_plots/png/fastp-seq-quality-plot_Read_2_Before_filtering.png",
                "multiqc/multiqc_plots/png/fastp_filtered_reads_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastp_filtered_reads_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/multiqc_plots/png/fastqc_sequence_length_distribution_plot.png",
                "multiqc/multiqc_plots/png/fastqc_top_overrepresented_sequences_table.png",
                "multiqc/multiqc_plots/png/general_stats_table.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/fastp-insert-size-plot.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-gc-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-content-n-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_1_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_1_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_2_After_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp-seq-quality-plot_Read_2_Before_filtering.svg",
                "multiqc/multiqc_plots/svg/fastp_filtered_reads_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastp_filtered_reads_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_sequence_length_distribution_plot.svg",
                "multiqc/multiqc_plots/svg/fastqc_top_overrepresented_sequences_table.svg",
                "multiqc/multiqc_plots/svg/general_stats_table.svg",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_pipeline_software_mqc_versions.yml",
                "samplesheet",
                "samplesheet/samplesheet.csv",
                "sim-data",
                "sim-data.csv",
                "sim-data/20230404-Bases2Fastq-Sim_QC.html",
                "sim-data/DefaultSample.fastp.html",
                "sim-data/DefaultSample.fastp.json",
                "sim-data/DefaultSample_1.fastp.fastq.gz",
                "sim-data/DefaultSample_1.fastp.fastq.gz.md5",
                "sim-data/DefaultSample_1.fastp.fastq.gz_fastqc_data.txt",
                "sim-data/DefaultSample_1.fastp.fastq.gz_fastqc_report.html",
                "sim-data/DefaultSample_1.fastp.fastq.gz_summary.txt",
                "sim-data/DefaultSample_2.fastp.fastq.gz",
                "sim-data/DefaultSample_2.fastp.fastq.gz.md5",
                "sim-data/DefaultSample_2.fastp.fastq.gz_fastqc_data.txt",
                "sim-data/DefaultSample_2.fastp.fastq.gz_fastqc_report.html",
                "sim-data/DefaultSample_2.fastp.fastq.gz_summary.txt",
                "sim-data/Metrics.csv",
                "sim-data/RunManifest.json",
                "sim-data/RunStats.json",
                "sim-data/Samples",
                "sim-data/Samples/DefaultProject",
                "sim-data/Samples/DefaultProject/DefaultSample",
                "sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_R1.fastq.gz",
                "sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_R2.fastq.gz",
                "sim-data/Samples/DefaultProject/DefaultSample/DefaultSample_stats.json",
                "sim-data/UnassignedSequences.csv"
            ],
            [
                "fastp-insert-size-plot.txt:md5,ac9a9c57ed6cbd69033968f727e755de",
                "fastp-seq-content-gc-plot_Read_1_After_filtering.txt:md5,fc7408bc90e9fd3768e041c470b297eb",
                "fastp-seq-content-gc-plot_Read_1_Before_filtering.txt:md5,f1b14d0b742540f46a447508b27c2a6d",
                "fastp-seq-content-gc-plot_Read_2_After_filtering.txt:md5,df0d79b9296d4ee8996f2a28ccff4b65",
                "fastp-seq-content-gc-plot_Read_2_Before_filtering.txt:md5,8f63cb5ddb7b4ebe8325271093018714",
                "fastp-seq-content-n-plot_Read_1_After_filtering.txt:md5,48fe699c2e7dfa81d9d97f53c613ef20",
                "fastp-seq-content-n-plot_Read_1_Before_filtering.txt:md5,48fe699c2e7dfa81d9d97f53c613ef20",
                "fastp-seq-content-n-plot_Read_2_After_filtering.txt:md5,48fe699c2e7dfa81d9d97f53c613ef20",
                "fastp-seq-content-n-plot_Read_2_Before_filtering.txt:md5,48fe699c2e7dfa81d9d97f53c613ef20",
                "fastp-seq-quality-plot_Read_1_After_filtering.txt:md5,e2194a8841653929cad0847f64daa339",
                "fastp-seq-quality-plot_Read_1_Before_filtering.txt:md5,559b29beffa82c35c71a82c3ff459a40",
                "fastp-seq-quality-plot_Read_2_After_filtering.txt:md5,62a344d18de885a880e8a9fe48c7816f",
                "fastp-seq-quality-plot_Read_2_Before_filtering.txt:md5,5ef421e1df317a359c8b27ff65f2b758",
                "fastp_filtered_reads_plot.txt:md5,5a5a98b82c02875b4d9da92e7577e802",
                "fastqc-status-check-heatmap.txt:md5,5fb90979330ac2f8ae13be85a68baff9",
                "fastqc_per_base_n_content_plot.txt:md5,7a1a9b3df3ce3bcca265dd13f2e23dc0",
                "fastqc_per_base_sequence_quality_plot.txt:md5,9f62301e59b368cdd6738cbd871a964e",
                "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,ebee4de078cbb125ba741a21d3552900",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,58b7050c7f02baa7fc308c691e62cc4c",
                "fastqc_per_sequence_quality_scores_plot.txt:md5,dd7520f2fd44bfb2fa67f8ff9f7237a5",
                "fastqc_sequence_counts_plot.txt:md5,e8772d59c78943686597222939492e1e",
                "fastqc_sequence_duplication_levels_plot.txt:md5,4b372fcce4dbb8b0dbe685c6380f4a97",
                "fastqc_sequence_length_distribution_plot.txt:md5,5b002b51364b518093ad98ebe0a0ca39",
                "fastqc_top_overrepresented_sequences_table.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "multiqc_citations.txt:md5,d35df50e9903a96a2b3bce3c1fbc8ad2",
                "multiqc_fastqc.txt:md5,733658661a4c44e230131be3f36e7edd",
                "multiqc_general_stats.txt:md5,7c21064d9b3913b1e26a94d53f24ebde",
                "sim-data.csv:md5,1d1dab697bd88c411fff89c102024295",
                "20230404-Bases2Fastq-Sim_QC.html:md5,288a791de5807768be54066239c50248",
                "DefaultSample.fastp.json:md5,9c10bbfbc4beb9523ca3a114f95709db",
                "DefaultSample_1.fastp.fastq.gz:md5,af3b00896a4b9a88a78d6483c77f8e41",
                "DefaultSample_1.fastp.fastq.gz.md5:md5,835c6ffa4d61d25036b70197b3248f1a",
                "DefaultSample_1.fastp.fastq.gz_fastqc_data.txt:md5,3ba5a3741834ef0247d4db40668eb7cf",
                "DefaultSample_1.fastp.fastq.gz_summary.txt:md5,8958835741049330b4b80d777b42c34c",
                "DefaultSample_2.fastp.fastq.gz:md5,bd01d7cdfcba45e0783377601dc57f2f",
                "DefaultSample_2.fastp.fastq.gz.md5:md5,05b248721c66e80a9426fc8b2cb47361",
                "DefaultSample_2.fastp.fastq.gz_fastqc_data.txt:md5,a7fb7fa1adab7d1ef03f9d3dcfa4b967",
                "DefaultSample_2.fastp.fastq.gz_summary.txt:md5,e134a7326ec678edcc3c6626ba7dde4d",
                "Metrics.csv:md5,ad7af77573845924062e2554cc641b16",
                "RunManifest.json:md5,cdb8051690645d0f070191286323f1c8",
                "RunStats.json:md5,bb40f4344d2f49c5cfb28c9d656c0ab1",
                "DefaultSample_R1.fastq.gz:md5,6c73b924e4eb53b716614aa6fd8b791a",
                "DefaultSample_R2.fastq.gz:md5,54e2c0894844e42605b7cd567b1c7dbe",
                "DefaultSample_stats.json:md5,37e555d1439d04f30e05d8028ec40cc4",
                "UnassignedSequences.csv:md5,11c1693830ce941b8cfb8d2431a59097"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T15:54:19.72597"
    }
}